pkg/pipeline/engine/engine.go

```diff
@@ -15,8 +15,9 @@ type PipelineEngine interface {
 	SyncExecution(execution *v3.PipelineExecution) (bool, error)
 }
 
-func New(cluster *config.UserContext) PipelineEngine {
-	serviceLister := cluster.Core.Services("").Controller().Lister()
+func New(cluster *config.UserContext, useCache bool) PipelineEngine {
+	services := cluster.Core.Services("")
```

Comment:
I think service should be fine with using cache. The service controller will start in follower in `https://github.com/rancher/rancher/blob/master/pkg/controllers/user/controllers.go#L88`

Reply:
So service cache is already added to followers in master, we only need to handle the rest(secrets in this case). Updated.